prime-angular-frontend/src/app/lib/modules/ngx-busy/busy.service.ts

```diff
@@ -18,13 +18,16 @@ export class BusyService {
   /**
    * @description
    * @param message the message to be displayed when a busy has a busy thing (subscription, promise, array of either... etc)
-   * @param request the observable that will run while the busy message and spinner are shown
+   * @param request the function or observable that will run while the busy message and spinner are shown
    * @returns OperatorFunction
    */
-  public showMessagePipe<T>(message: string, request: Observable<any>): OperatorFunction<T, Observable<any>> {
+  public showMessagePipe<T, R>(message: string, request: R): UnaryFunction<Observable<T>, Observable<any>> {
```

Comment:
What's preventing me from passing a number, object literal, or anything else into this method, or passing a method that returns anything I want?  The typings don't indicate what is expected and the generic is wide open.  Only your code does a small check for a function and as it is your `exhaustMap` will fail since it expects an observable to be returned.  This is why I mentioned you should extend the signature of your parameters to include what you expect.

Reply:
Also, is the UnaryFunction entering with a type of `Observable<T>`?

Reply:
Return type was an oversight from before.